dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -261,13 +261,8 @@ private String resultProcess(ResultSet resultSet) throws Exception {
                 resultJSONArray.add(mapOfColValues);
                 rowCount++;
             }
-            int displayRows = sqlParameters.getDisplayRows() > 0 ? sqlParameters.getDisplayRows() : TaskConstants.DEFAULT_DISPLAY_ROWS;
-            displayRows = Math.min(displayRows, rowCount);
-            logger.info("display sql result {} rows as follows:", displayRows);
-            for (int i = 0; i < displayRows; i++) {
-                String row = JSONUtils.toJsonString(resultJSONArray.get(i));
-                logger.info("row {} : {}", i + 1, row);
-            }
+            // generate query results
+            generateRow(resultJSONArray, md, num, rowCount);
         }
         String result = JSONUtils.toJsonString(resultJSONArray);
         if (sqlParameters.getSendEmail() == null || sqlParameters.getSendEmail()) {
```

Comment:
Avoid changing the return result.
```suggestion
        if (sqlParameters.getSendEmail() == null || sqlParameters.getSendEmail()) {
            String attachmentContent = resultJSONArray.isEmpty()? JSONUtils.toJsonString(generateEmptyRow()):result;
            sendAttachment(sqlParameters.getGroupId(), StringUtils.isNotEmpty(sqlParameters.getTitle())
                    ? sqlParameters.getTitle()
                    : taskExecutionContext.getTaskName() + " query result sets", attachmentContent);
```